R/CITEseq.R

```diff
@@ -1,18 +1,145 @@
+                                            names(explist)[assIdx[1]]),
+                                        dimslist[assIdx][[1]][2]), 
+                                    colnames(explist[[assIdx[1]]])),
+                              paste0(rep(gsub("scADT|scHTO|scRNA","", 
+                                            names(explist)[assIdx[2]]),
+                                        dimslist[assIdx][[2]][2]), 
+                                    colnames(explist[[assIdx[2]]])))
+        } else {
+            colnames(m1) <- paste0(rep(gsub("scADT|scHTO|scRNA","", 
+                                            names(explist)[assIdx[1]]),
+                                        dimslist[assIdx][[1]][2]), 
+                                    colnames(explist[[assIdx[1]]]))
+        }
+        return(m1)
+    }
+    
+    .buildMap <- function(mat1, assayId)
+    {
+        
+        map <- DataFrame(assay=assayId,
+                        #primary=gsub("_\\w+", "", colnames(mat1)), 
+                        primary=colnames(mat1),
+                        colname=colnames(mat1), 
+                        condition=gsub("_\\w+", "", colnames(mat1)))
+        return(map)
+    }
+    
+    # .buildColDat <- function(ll)
+    # {
+    #     if(all(names(ll)[grep("CTCL|CTRL", names(ll))] %in% names(ll)))
+    #     {
+    #         cd <- data.frame(row.names=c("CTCL", "CTRL"), 
+    #                    condition=c("Cutaneous T-cell Limphoma", "Control"))
+    #     } else if(!isEmpty(grep("CTCL", names(ll)))) {
+    #         cd <- data.frame(row.names=c("CTCL"), 
+    #                           condition=c("Cutaneous T-cell Limphoma"))
+    #     } else if(!isEmpty(grep("CTRL", names(ll)))) {
+    #         cd <- data.frame(row.names=c("CTRL"), 
+    #                           condition=c("Control"))
+    #     }
+    #     return(cd)
+    # }
+    
+    ll <- ess_list$experiments
+    ll <- lapply(ll, function(x) 
+    {
+        x <- x[order(rownames(x)),]
+    })
+    dims <- lapply(ll, dim)
+    expslist <- c()
+    sampmap <- DataFrame()
+    if(!isEmpty(grep("scADT", names(ll))))
+    {
+        ADTs <- .combMatrixForAssay(explist=ll, dimslist=dims, assayId="scADT")
+        expslist <- c(expslist, scADT=ADTs)
+        ADTsMap <- .buildMap(ADTs, assayId="scADT")
+        sampmap <- rbind(sampmap, ADTsMap)
+    }
+    if(!isEmpty(grep("scHTO", names(ll))))
+    {
+        HTOs <- .combMatrixForAssay(explist=ll, dimslist=dims, assayId="scHTO")
+        expslist <- c(expslist, scHTO=HTOs)
+        HTOsMap <- .buildMap(HTOs, assayId="scHTO")
+        sampmap <- rbind(sampmap, HTOsMap)
+    }
+    if(!isEmpty(grep("scRNA", names(ll))))
+    {
+        RNAs <- .combMatrixForAssay(explist=ll, dimslist=dims, assayId="scRNA")
+        expslist <- c(expslist, scRNA=RNAs)
+        RNAsMap <- .buildMap(RNAs, assayId="scRNA")
+        sampmap <- rbind(sampmap, RNAsMap)
+    }
```

Comment:
Lines 99 - 120 can be combined into one operation

Reply:
what do you mean with "one operation"?